client/repodlgs_common.cpp

```diff
@@ -150,6 +150,63 @@ void get_economy_report_units_data(struct unit_entry *entries,
   unit_type_iterate_end;
 }
 
+/**
+   Returns an array of units data. Number of units in
+   the array is added to num_entries_used.
+ */
+void get_units_view_data(struct unit_view_entry *entries,
+                         int *num_entries_used)
+{
+  int count, gold_cost, food_cost, shield_cost;
```

Comment:
Best declare variables on first use (reduces the number of variables one needs to track when reading the code).